Source/NSSortDescriptor.m

```diff
@@ -113,16 +113,31 @@ - (NSComparisonResult) compareObject: (id) object1 toObject: (id) object2
   id comparedKey1 = [object1 valueForKeyPath: _key];
   id comparedKey2 = [object2 valueForKeyPath: _key];
 
-  if (_comparator == NULL)
-    {
-      result = (NSComparisonResult) [comparedKey1 performSelector: _selector
-                                                       withObject: comparedKey2];
+  // Cocoa seems to either capture the exception thrown in the nil parameter case or
```

Comment:
This looks like a hack to me. What is nil here and isn't there any better way to handle that case?
The solution to always return NSOrderedAscending regardless of the arguments might result in some sorting algorithms to get into an endless loop.